BizHawk.Emulation.Common/Interfaces/IMemoryCallbackSystem.cs

```diff
@@ -121,4 +112,20 @@ public enum MemoryCallbackType
 	{
 		Read, Write, Execute
 	}
+
+	[Flags]
+	public enum MemoryCallbackFlags : uint
+	{
+		SizeUnknown = 0x00 << 16,
+		SizeByte = 0x01 << 16,
+		SizeWord = 0x02 << 16,
+		SizeLong = 0x03 << 16,
+		AccessUnknown = 0x00 << 12,
+		AccessRead = 0x01 << 12,
+		AccessWrite = 0x02 << 12,
+		AccessExecute = 0x04 << 12,
+		CPUUnknown = 0x00 << 8,
+		CPUZero = 0x01 << 8,
+		DomainUnknown = 0x00,
+	}

```

Comment:
Is there a preference to order enums? Even though numbers are written from highest to lowest digit, I'd personally prefer enums to go from lowest to highest.

Reply:
also shouldn't this be a struct with 4 byte fields (or enums extending byte)

Reply:
Two of the fields are only a nibble wide. I think in general, the [Flags] style enum is pretty clean since you can use .HasFlag and use bitwise OR

Reply:
Though thinking about it more, I realize that HasFlag will report incorrectly for sizes with the way I wrote this. Maybe struct is the way to go

Reply:
This is designed for efficiency, not code niceness. I dont know what you mean by "a struct" but this should stay a uint. Make utilities somewhere to extract the fields from the packed uint.

Reply:
I don't know if we really need utilities, .HasFlag covers most of the cases, and otherwise people can just do a little bit of bit manip

Reply:
[what I meant](https://gist.github.com/YoshiRulz/2224fbfba9960b0a4b236d8962cf9f2f)
don't copy-paste that without checking the attributes though, the docs are lacking

Reply:
Weird, I could have sworn I posted this earlier but I cant find it now. HasFlag has bad performance. Don't like yoshirulz four-fields-only approach. You guys are overthinking this. If you want to overthink it, add some extension methods for unpacking the fields right by the enum so nobody can miss it.